### PR TITLE
Changes model field `diffusivities` to `diffusivity_fields`

### LINKED ISSUE #1878: `NonhydrostaticModel` creates an extra empty element in `diffusivities` list when a tuple is passed to as `closure`
I'm assuming the behavior below isn't expected., where `NonhydrostaticModel` creates an element in `diffusivities` for each element in a tuple of closures but one of them is empty. What would be the preferred way?

```julia
julia> using Oceananigans

julia> grid = RegularRectilinearGrid(size=(4, 4, 4), extent=(1, 1, 1), topology=(Periodic, Periodic, Bounded))
RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}
                   domain: x ∈ [0.0, 1.0], y ∈ [0.0, 1.0], z ∈ [-1.0, 0.0]
                 topology: (Periodic, Periodic, Bounded)
  resolution (Nx, Ny, Nz): (4, 4, 4)
   halo size (Hx, Hy, Hz): (1, 1, 1)
grid spacing (Δx, Δy, Δz): (0.25, 0.25, 0.25)

julia> closure = (AnisotropicMinimumDissipation(), IsotropicDiffusivity(ν=1e-6, κ=1.4e-7))
(AnisotropicMinimumDissipation{Float64} turbulence closure with:
           Poincaré constant for momentum eddy viscosity Cν: 0.08333333333333333
    Poincaré constant for tracer(s) eddy diffusivit(ies) Cκ: 0.08333333333333333
                        Buoyancy modification multiplier Cb: nothing
                Background diffusivit(ies) for tracer(s), κ: 0.0
             Background kinematic viscosity for momentum, ν: 0.0, IsotropicDiffusivity: ν=1.0e-6, κ=1.4e-7)

julia> model = NonhydrostaticModel(grid=grid, closure=closure)
NonhydrostaticModel{CPU, Float64}(time = 0 seconds, iteration = 0) 
├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=4, Ny=4, Nz=4)
├── tracers: (:T, :S)
├── closure: Tuple{AnisotropicMinimumDissipation{Oceananigans.TurbulenceClosures.ExplicitTimeDiscretization, Float64, NamedTuple{(:T, :S), Tuple{Float64, Float64}}, Float64, NamedTuple{(:T, :S), Tuple{Float64, Float64}}, Nothing}, IsotropicDiffusivity{Oceananigans.TurbulenceClosures.ExplicitTimeDiscretization, Float64, NamedTuple{(:T, :S), Tuple{Float64, Float64}}}}
├── buoyancy: SeawaterBuoyancy{Float64, LinearEquationOfState{Float64}, Nothing, Nothing}
└── coriolis: Nothing

julia> model.diffusivities[1]
(νₑ = Field located at (Center, Center, Center)
├── data: OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, size: (4, 4, 4)
├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=4, Ny=4, Nz=4)
└── boundary conditions: west=Periodic, east=Periodic, south=Periodic, north=Periodic, bottom=ZeroFlux, top=ZeroFlux, immersed=ZeroFlux, κₑ = (T = Field located at (Center, Center, Center)
├── data: OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, size: (4, 4, 4)
├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=4, Ny=4, Nz=4)
└── boundary conditions: west=Periodic, east=Periodic, south=Periodic, north=Periodic, bottom=ZeroFlux, top=ZeroFlux, immersed=ZeroFlux, S = Field located at (Center, Center, Center)
├── data: OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, size: (4, 4, 4)
├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=4, Ny=4, Nz=4)
└── boundary conditions: west=Periodic, east=Periodic, south=Periodic, north=Periodic, bottom=ZeroFlux, top=ZeroFlux, immersed=ZeroFlux))

julia> model.diffusivities[2]

```

Note that the output of the last line is empty. So the second element is there, but it's `nothing`. If I check the model's property closure, I can see both closures though:

```julia
julia> model.closure[1]
AnisotropicMinimumDissipation{Float64} turbulence closure with:
           Poincaré constant for momentum eddy viscosity Cν: 0.08333333333333333
    Poincaré constant for tracer(s) eddy diffusivit(ies) Cκ: (T = 0.08333333333333333, S = 0.08333333333333333)
                        Buoyancy modification multiplier Cb: nothing
                Background diffusivit(ies) for tracer(s), κ: (T = 0.0, S = 0.0)
             Background kinematic viscosity for momentum, ν: 0.0

julia> model.closure[2]
IsotropicDiffusivity: ν=1.0e-6, κ=(T = 1.4e-7, S = 1.4e-7)
```

So this is a bit confusing (at least to me) and I guess it relates to https://github.com/CliMA/Oceananigans.jl/issues/1381.

Thoughts?

### PR DESCRIPTION
**Pull request by tomchor**
Closes #1878

I ran the most relevant tests locally and they all passed for me (didn't have time/resources to run all), but I'd appreciate some thorough reviews in this case since "diffusivities" is a word that happens a lot in the repo so it's easy to miss something when making such a change.

### CODE REVIEW

**Comment by tomchor** [comment]
For posterity, here's the command I used to do the heavy-lifting:

```
find . -type f -exec sed -i "s/\.diffusivities/\.diffusivity_fields/g" {} \;
```

File: src/Models/NonhydrostaticModels/nonhydrostatic_model.jl [hunk 1]
```diff
@@ -148,7 +148,7 @@ function NonhydrostaticModel(;    grid,
     velocities    = VelocityFields(velocities, architecture, grid, boundary_conditions)
     tracers       = TracerFields(tracers,      architecture, grid, boundary_conditions)
     pressures     = PressureFields(pressures,  architecture, grid, boundary_conditions)
-    diffusivities = DiffusivityFields(diffusivities, architecture, grid, tracernames(tracers), boundary_conditions, closure)
+    diffusivity_fields = DiffusivityFields(diffusivity_fields, architecture, grid, tracernames(tracers), boundary_conditions, closure)
```
Comment (glwagner) [inline src/Models/NonhydrostaticModels/nonhydrostatic_model.jl:151]: The formatting got mangled here, please fix.

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl [hunk 2]
```diff
@@ -42,7 +42,7 @@ mutable struct HydrostaticFreeSurfaceModel{TS, E, A<:AbstractArchitecture, S,
           velocities :: U        # Container for velocity fields `u`, `v`, and `w`
              tracers :: C        # Container for tracer fields
             pressure :: Φ        # Container for hydrostatic pressure
-       diffusivities :: K        # Container for turbulent diffusivities
+  diffusivity_fields :: K        # Container for turbulent diffusivities
          timestepper :: TS       # Object containing timestepper fields and parameters
     auxiliary_fields :: AF       # User-specified auxiliary fields for forcing functions and boundary conditions
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl:47]: The formatting got mangled (the longest name, which is now `diffusivity_fields`, should be indented by 4). These 20 lines or so need to be indented by 2 more spaces (please visually inspect to ensure it is readable).

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl [hunk 3]
```diff
@@ -148,7 +148,7 @@ function HydrostaticFreeSurfaceModel(; grid,
     velocities    = HydrostaticFreeSurfaceVelocityFields(velocities, architecture, grid, clock, boundary_conditions)
     tracers       = TracerFields(tracers, architecture, grid, boundary_conditions)
     pressure      = PressureField(architecture, grid)
-    diffusivities = DiffusivityFields(diffusivities, architecture, grid, tracernames(tracers), boundary_conditions, closure)
+    diffusivity_fields = DiffusivityFields(diffusivity_fields, architecture, grid, tracernames(tracers), boundary_conditions, closure)
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl:151]: Formatting is mangled, please fix.

File: src/Models/NonhydrostaticModels/nonhydrostatic_model.jl [hunk 4]
```diff
@@ -36,7 +36,7 @@ mutable struct NonhydrostaticModel{TS, E, A<:AbstractArchitecture, G, T, B, R, S
            velocities :: U        # Container for velocity fields `u`, `v`, and `w`
               tracers :: C        # Container for tracer fields
             pressures :: Φ        # Container for hydrostatic and nonhydrostatic pressure
-        diffusivities :: K        # Container for turbulent diffusivities
+   diffusivity_fields :: K        # Container for turbulent diffusivities
           timestepper :: TS       # Object containing timestepper fields and parameters
       pressure_solver :: S        # Pressure/Poisson solver
     immersed_boundary :: I        # Models the physics of immersed boundaries within the grid
```
Comment (glwagner) [inline src/Models/NonhydrostaticModels/nonhydrostatic_model.jl:42]: Like for the hydrostatic model, the formatting of the model properties is mangled.

File: src/Models/ShallowWaterModels/shallow_water_model.jl [hunk 5]
```diff
@@ -78,7 +78,7 @@ Keyword arguments
     - `bathymetry`: The bottom bathymetry.
     - `tracers`: A tuple of symbols defining the names of the modeled tracers, or a `NamedTuple` of
                  preallocated `CenterField`s.
-    - `diffusivities`: 
+    - `diffusivity_fields`:
```
Comment (glwagner) [inline src/Models/ShallowWaterModels/shallow_water_model.jl:81]: There's no description of this argument --- might be good to add.

File: src/Models/ShallowWaterModels/shallow_water_model.jl [hunk 6]
```diff
@@ -113,7 +113,7 @@ function ShallowWaterModel(;
 
     solution = ShallowWaterSolutionFields(architecture, grid, boundary_conditions)
     tracers  = TracerFields(tracers, architecture, grid, boundary_conditions)
-    diffusivities = DiffusivityFields(diffusivities, architecture, grid,
+    diffusivity_fields = DiffusivityFields(diffusivity_fields, architecture, grid,
                                       tracernames(tracers), boundary_conditions, closure)
```
Comment (glwagner) [inline src/Models/ShallowWaterModels/shallow_water_model.jl:117]: The formatting of these two lines is mangled.

File: validation/stratified_couette_flow/stratified_couette_flow.jl [hunk 7]
```diff
@@ -196,8 +196,8 @@ function simulate_stratified_couette_flow(; Nxy, Nz, arch=GPU(), h=1, U_wall=1,
     Vavg = AveragedField(model.velocities.v,       dims=(1, 2))
     Wavg = AveragedField(model.velocities.w,       dims=(1, 2))
     Tavg = AveragedField(model.tracers.T,          dims=(1, 2))
-    νavg = AveragedField(model.diffusivities.νₑ,   dims=(1, 2))
-    κavg = AveragedField(model.diffusivities.κₑ.T, dims=(1, 2))
+    νavg = AveragedField(model.diffusivity_fields.νₑ,   dims=(1, 2))
+    κavg = AveragedField(model.diffusivity_fields.κₑ.T, dims=(1, 2))
```
Comment (glwagner) [inline validation/stratified_couette_flow/stratified_couette_flow.jl:200]: Formatting is mangled here. Please fix.

**Comment by glwagner** [comment]
The change is great, but the code needs to be proofread to remove formatting inconsistencies. I may not have caught all of them, so I think it'd be good to look over every file individually. In the future this is probably also a better way to search and replace rather than issuing a single command (except in the lucky case where the replaced pattern has the same number of characters as the original one, like incompressible -> nonhydrostatic).

**Review by glwagner — APPROVED** [review]

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl (hunk 2 above)
Comment (tomchor) [reply src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl:47]: I'll do it, but just so I know: why is 4 the minimum indentation? Does it have to do with Documenter.jl when rendering?

Comment (glwagner) [reply src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl:47]: No, it's just our convention to use 4 space for indents. It has no effect on execution.

**Comment by tomchor** [comment]
> It may be more efficient in the future to issue search and replace in individual files, so the change sites can be inspected at that time (except in lucky cases where the replaced pattern has the same number of characters as the original one, like incompressible -> nonhydrostatic).

I actually did that and fixed most of the formatting by hand. The thing is that I think you're way more attentive to the formatting than I am :joy:, so there were some things that I completely missed and others that I saw but thought were fine. 

For example the 4-col indentation for arguments, which I didn't know was a thing. The way that Oceananigans indents some stuff (by aligning the equals sign) is very unique and I had never seen it before so I may not have caught all the rules yet. (I always use the [python standard](https://www.python.org/dev/peps/pep-0008/) for all my code, which is easier to enforce and automatize I think.)

**Comment by tomchor** [comment]
Gave it a couple more passes and didn't find anything badly formatted, so I'm merging this

**Pull request merged**